corehq/apps/export/urls.py

```diff
@@ -65,7 +65,7 @@
     url(r"^custom/download_data_files/$",
         DataFileDownloadList.as_view(),
         name=DataFileDownloadList.urlname),
-    url(r"^custom/download_data_files/(?P<pk>\d+)/(?P<filename>.*)$",
+    url(r"^custom/download_data_files/(?P<pk>.*)/(?P<filename>.*)$",
```

Comment:
Why did this need to change?

Reply:
I changed this because I wanted to `register` this url. and revsering the url is made to only replace '---'(str) with the actual value [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/static/hqwebapp/js/initial_page_data.js#L99). could have updated to support the numbers, But that I has already been for used for integers in url like at number of places. changing that would need to change all those places

Reply:
Ah, interesting, I hadn't thought about `registerurl` having that limitation. It's not ideal that you have to change this URL to accept non-digits just because of the js, but I don't see a better simple solution. Although you might consider switching from `.*` to `[0-9-]` or the common `[\w\-]+` rather than accepting absolutely anything - it's good to make regexes restrictive.